pilot/pkg/xds/eds.go

```diff
@@ -470,25 +522,14 @@ func (s *DiscoveryServer) pushEds(push *model.PushContext, con *Connection, vers
 	return nil
 }
 
-// getDestinationRule gets the DestinationRule for a given hostname. As an optimization, this also gets the service port,
-// which is needed to access the traffic policy from the destination rule.
-func getDestinationRule(push *model.PushContext, proxy *model.Proxy, hostname host.Name) *networkingapi.DestinationRule {
-	cfg := push.DestinationRule(proxy, push.ServiceForHostname(proxy, hostname))
-	if cfg != nil {
-		return cfg.Spec.(*networkingapi.DestinationRule)
+func getOutlierDetectionAndLoadBalancerSettings(cfg *model.Config, portNumber int, subsetName string) (bool, *networkingapi.LoadBalancerSettings) {
+	if cfg == nil {
+		return false, nil
 	}
-	return nil
-}
-
-func getOutlierDetectionAndLoadBalancerSettings(push *model.PushContext, proxy *model.Proxy, clusterName string) (bool, *networkingapi.LoadBalancerSettings) {
-	_, subsetName, hostname, portNumber := model.ParseSubsetKey(clusterName)
+	destinationRule := cfg.Spec.(*networkingapi.DestinationRule)
 	var outlierDetectionEnabled = false
 	var lbSettings *networkingapi.LoadBalancerSettings
 
-	destinationRule := getDestinationRule(push, proxy, hostname)
-	if destinationRule == nil {
-		return false, nil
-	}
```

Comment:
I think this should not be removed?

Reply:
Its moved to L526, `cfg == nil`. `destinationRule` in the old code is equivalent to cfg, `destinationRule` in new code cannot be nil.

Reply:
Ok. Got it.